### ISSUE TITLE
x/build: reverse builder config uniformity

### ISSUE THREAD

**Issue by adams-sarah** [golang/go#21260]
Onboarding overhead is very high ATM for builder project.

I'd like to discuss a config uniformity overhaul for the reverse buildlets.
We have been migrating various reverse buildlet Dockerfiles / configurations over to various different stages of "new" piecemeal b/c lack of headcount. eg. uses X version of rundockerbuildlet, uses rundockerbuildlet, are builds hermetic?, uses `--reverse` (legacy) vs. `--reverse-type` (latest) flag to `cmd/buildlet/buildlet.go`

I'd like us document the ideal/latest blessed reverse buildlet config, and push to bring all reverse buildlet configurations up to this config scheme.
I'd also like to discuss the potential for a plan to keep reverse buildlet configurations at `latest` config scheme in the future.

**Comment by bradfitz** [comment]
I will start referencing bugs from here.

Here is why the Solaris/illumos builders are not ideal: #15581

**Comment by bradfitz** [comment]
I deployed CL 52633 to see which hosts are using the old `--reverse=` argument to `buildlet` instead of `--reverse-type=` and found:

```
      1 host type host-darwin-10_8 (mapped from legacy modes ["darwin-amd64-10_8"])
      1 host type host-linux-mipsle (mapped from legacy modes ["linux-mipsle"])
      1 host type host-linux-mips (mapped from legacy modes ["linux-mips"])
      1 host type host-linux-s390x (mapped from legacy modes ["linux-s390x-ibm"])
      2 host type host-darwin-10_12 (mapped from legacy modes ["darwin-amd64-10_12"])
      2 host type host-solaris-amd64 (mapped from legacy modes ["solaris-amd64-smartosbuildlet"])
      3 host type host-darwin-10_10 (mapped from legacy modes ["darwin-amd64-10_10"])
      3 host type host-linux-arm5spacemonkey (mapped from legacy modes ["linux-arm-arm5spacemonkey"])
      5 host type host-linux-ppc64le-osu (mapped from legacy modes ["linux-ppc64le-buildlet"])
      5 host type host-linux-ppc64-osu (mapped from legacy modes ["linux-ppc64-buildlet"])
     14 host type host-darwin-10_11 (mapped from legacy modes ["darwin-amd64-10_11"])
     50 host type host-linux-arm-scaleway (mapped from legacy modes ["host-linux-arm-scaleway"])
```

Updating the Scaleway images can happen in https://go-review.googlesource.com/c/52293

A bunch of the others are done in https://go-review.googlesource.com/c/52650 (once ppc64 and arm5-spacemonkey are updated)

Updating Mac images is super hard and tedious and SLOW and error prone. I think we'd be better of keeping the mapping in the coordinator and buildlet for a bit. We can hide it from flag listing if helpful to minimize doc clutter.

MIPS seems to be updateable easily enough.

I also want to figure out a plan for #21191 (speaking of cleanup).

**Comment by bradfitz** [comment]
Latest update on which hosts are still using the legacy ways:

* host type host-darwin-10_10 (mapped from legacy modes ["darwin-amd64-10_10"])
* host type host-darwin-10_11 (mapped from legacy modes ["darwin-amd64-10_11"])
* host type host-darwin-10_12 (mapped from legacy modes ["darwin-amd64-10_12"])
* host type host-linux-arm-scaleway (mapped from legacy modes ["host-linux-arm-scaleway"])
* host type host-linux-ppc64-osu (mapped from legacy modes ["linux-ppc64-buildlet"])
* host type host-linux-s390x (mapped from legacy modes ["linux-s390x-ibm"])
* host type host-solaris-amd64 (mapped from legacy modes ["solaris-amd64-smartosbuildlet"])

Basically no change since Aug 1 2017, except that we no longer OS X 10.8 and the MIPS builders disappeared.

The one fix seems to be that host-linux-ppc64le-osu and host-linux-ppc64-osu are no longer doing the legacy registration.

Also, macOS 10.14 is at least using the new way. And the Mac VMware cluster is easier to work with now since it was upgraded, so we could fix those old versions at least.

**Comment by bradfitz** [comment]
Two of those builders are gone now, so remaining is:

* host type host-darwin-10_11 (mapped from legacy modes ["darwin-amd64-10_11"])
* host type host-darwin-10_12 (mapped from legacy modes ["darwin-amd64-10_12"])
* host type host-linux-arm-scaleway (mapped from legacy modes ["host-linux-arm-scaleway"])
* host type host-linux-ppc64-osu (mapped from legacy modes ["linux-ppc64-buildlet"])
* host type host-linux-s390x (mapped from legacy modes ["linux-s390x-ibm"])

**Comment by bradfitz** [comment]
Down to:

* host type host-darwin-10_11 (mapped from legacy modes ["darwin-amd64-10_11"])
* host type host-darwin-10_12 (mapped from legacy modes ["darwin-amd64-10_12"])
* host type host-linux-arm-scaleway (mapped from legacy modes ["host-linux-arm-scaleway"])
* host type host-linux-s390x (mapped from legacy modes ["linux-s390x-ibm"])

**Comment by bradfitz** [comment]
Just double checked, and yes:

```
$ kubectl logs $(k get -o name po -l app=coordinator) | grep "mapped from legacy" | perl -ne 'm/host type (\S+)/ and print "$1\n"' | sort | uniq -c
     29 host-darwin-10_11
    215 host-darwin-10_12
    327 host-linux-arm-scaleway 
    214 host-linux-s390x
```

**Comment by bradfitz** [comment]
And I just fixed the darwin ones.
